mypy/newsemanal/semanal_shared.py

```diff
@@ -122,7 +123,18 @@ def schedule_patch(self, priority: int, fn: Callable[[], None]) -> None:
 
     @abstractmethod
     def add_symbol_table_node(self, name: str, stnode: SymbolTableNode) -> bool:
-        """Add node to global symbol table (or to nearest class if there is one)."""
+        """Add node to the current symbol table."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def add_symbol(self, name: str, node: SymbolNode, context: Optional[Context],
+                   module_public: bool = True, module_hidden: bool = False) -> bool:
+        """Add symbol to the current symbol table."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def add_symbol_skip_local(self, name: str, node: SymbolNode) -> None:
+        """Add symbol to the current symbol table, skipping locals."""
```

Comment:
Explain the purpose of this method, since it's not obvious why we'd need something like this.